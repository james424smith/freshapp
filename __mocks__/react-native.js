import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
