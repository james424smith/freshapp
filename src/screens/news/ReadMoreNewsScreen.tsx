import React from "react";
import { View } from "react-native";
import constants from "../../constants/config";
import WebRender from "../../components/WebRender";
import styles, { stylesheet } from "../../styles/more_screens/NewsCardStyles";
import { NEWSLETTERS } from "../../constants/NewsConstants";
import { AssignmentStackParamList } from "../../navigation/MainTabNavigator";
import { RouteProp } from "@react-navigation/native";
import PdfRender from "../../components/PdfRender";
import Colors from "../../constants/styles/colors";

interface Props {
  route: RouteProp<AssignmentStackParamList, "LatestReleases">;
}

export default (props: Props) => {
  const { route } = props;
  const imagePrefix = constants().IMAGES_PREFIX_URL;

  const newItem = route.params?.newItem;
  return newItem.newItemType === NEWSLETTERS ? (
    <PdfRender document={newItem.document} resourceType={"url"} />
  ) : (
    <View
      style={{
        ...styles.containerWebRender,
        backgroundColor: Colors.white,
      }}
    >
      <WebRender
        stringToRender={`<html>${stylesheet}<body><img <href src="${imagePrefix}/${newItem?.imageUrl}" />${newItem?.text}<body></html>`}
        resourceType={"html"}
      />
    </View>
  );
};
