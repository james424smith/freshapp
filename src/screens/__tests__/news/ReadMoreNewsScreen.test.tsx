import React from "react";
import { render, cleanup } from "@testing-library/react-native";
import ReadMoreNewsScreen from "../../news/ReadMoreNewsScreen";

describe("ReadMoreNewsScreen snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render ReadMoreNewsScreen when loader is false when newItemType is NEWSLETTERSn", () => {
    const route = {
      state: {},
      key: "LatestReleases-E0ArYKZZNpvIFYMFIEyaQ",
      name: "LatestReleases" as "LatestReleases",
      params: {
        newItem: {
          imageUrl: "google.com",
          text: "1234",
          newItemType: "Newsletters",
          document: "1234",
        },
      },
      getState: jest.fn(),
    };

    const { toJSON } = render(<ReadMoreNewsScreen route={route} />);

    expect(toJSON()).toMatchSnapshot();
  });
  it("should render ReadMoreNewsScreen when loader is false when newItemType is PressRelease", () => {
    const route = {
      state: {},
      key: "LatestReleases-E0ArYKZZNpvIFYMFIEyaQ",
      name: "LatestReleases" as "LatestReleases",
      params: {
        newItem: {
          imageUrl: "google.com",
          text: "1234",
          newItemType: "PRESS_RELEASE",
          document: "1234",
        },
      },
      getState: jest.fn(),
    };

    const { toJSON } = render(<ReadMoreNewsScreen route={route} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
