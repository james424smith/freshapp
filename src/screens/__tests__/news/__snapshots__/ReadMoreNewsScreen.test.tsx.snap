// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadMoreNewsScreen snapshot test should render ReadMoreNewsScreen when loader is false when newItemType is NEWSLETTERSn 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <PDFView
    fadeInDuration={10}
    onError={[Function]}
    onLoad={[Function]}
    onPageChanged={[Function]}
    onScrolled={[Function]}
    resource="1234"
    resourceType="url"
    style={
      Object {
        "flex": 1,
        "width": 750,
      }
    }
    textEncoding="utf-8"
    urlProps={Object {}}
  />
</View>
`;

exports[`ReadMoreNewsScreen snapshot test should render ReadMoreNewsScreen when loader is false when newItemType is PressRelease 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexGrow": 1,
      "paddingBottom": 22.5,
      "paddingLeft": 22.5,
      "paddingRight": 22.5,
      "paddingTop": 37.5,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      source={
        Object {
          "html": "<html><head>
	<link href='http://fonts.googleapis.com/css?family=Roboto' rel='stylesheet' type='text/css'>
	<style>
        body {
            font-family: 'Roboto';
        }

        p {
            font-size: 30;
            font-family: 'Roboto';
        }

        li {
            font-size: 25;
            font-family: 'Roboto';
        }
    </style>
<head><body><img <href src=\\"undefined/google.com\\" />1234<body></html>",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          undefined,
        ]
      }
      useSharedProcessPool={true}
    />
  </View>
</View>
`;
